classes/AlertView/Standard.php

```diff
@@ -438,10 +438,22 @@ private function searchForConstituenciesAndMembers() {
 
         if (isset($this->data['constituencies'])) {
             $cons = [];
-            foreach ($this->data['constituencies'] as $constituency) {
+            foreach ($this->data['constituencies'] as $type => $constituency) {
```

Comment:
But isn't this all overridden by the member stuff? There's probably things about the canonical constituency names that mean it isn't, I guess, whereby something you search for is an alternative name which would then match constituency but no members. But this seems very convoluted - should we not a) canonicalise constituency name, the b) do member lookups - rather than both?